src/System.Private.CoreLib/shared/System/Number.Formatting.cs

```diff
@@ -243,8 +243,8 @@ namespace System
     internal static partial class Number
     {
         internal const int DecimalPrecision = 29; // Decimal.DecCalc also uses this value
-        private const int SinglePrecision = 7;
-        private const int DoublePrecision = 15;
+        private const int SinglePrecision = 9;
```

Comment:
For the uninitiated, can you please add a comment explaining how we got to these constant values and why they are changing?

Reply:
Do you mean in source code or here in the PR?

The numbers just represent the maximum number of digits required to guarantee that any given `float` or `double` can roundtrip. Some numbers may require less, but no number will require more than 9 or 17, respectively.

Reply:
Because 7 and 15 aren't enough to guarantee roundtripping. They basically represent the "maximum" number of digits we need to return.

Just calling `ToString()` sets the precision to `-1` which computes the shortest roundtrippable string. The number of digits to display would then normally be computed as `DigitCount`, which in the case of a number like `-60` is actually 1 and we would print `-6E1`. In the case of the shortest roundtrippable string, we instead want to say we need no more than 9/17 digits so that we will print the better number `-60`.

This is explained in more detail at the consumption site of these constants here: https://github.com/dotnet/coreclr/pull/22040/files#diff-37631fe33cc7e94eacd4f41d57c11ed6R554